### ISSUE TITLE
Error when installing the environment

### ISSUE THREAD

**Issue by hangzhiyiwei** [JacobYuan7/RLIP#7]
Hi, thank you for your nice work.

When I run the inference code 

"bash scripts/Fine-tune_RLIP-ParSe_HICO.sh"

It shows the error

  File "/home/qinqian/RLIP/models/hoi.py", line 24, in <module>
    from .transformer import TransformerEncoder, TransformerEncoderLayer
  File "/home/qinqian/RLIP/models/transformer.py", line 29, in <module>
    from .ParSetransformer import ParSeTransformer, ParSeDeformableTransformer
  File "/home/qinqian/RLIP/models/ParSetransformer.py", line 23, in <module>
    from transformers import RobertaModel, RobertaTokenizerFast, BertTokenizerFast, BertModel
  File "/home/qinqian/anaconda3/envs/augm/lib/python3.8/site-packages/transformers/__init__.py", line 43, in <mod
ule>
    from . import dependency_versions_check
  File "/home/qinqian/anaconda3/envs/augm/lib/python3.8/site-packages/transformers/dependency_versions_check.py",
 line 41, in <module>
    require_version_core(deps[pkg])  
  File "/home/qinqian/anaconda3/envs/augm/lib/python3.8/site-packages/transformers/utils/versions.py", line 101, 
in require_version_core
    return require_version(requirement, hint)
  File "/home/qinqian/anaconda3/envs/augm/lib/python3.8/site-packages/transformers/utils/versions.py", line 92, i
n require_version
    if want_ver is not None and not ops[op](version.parse(got_ver), version.parse(want_ver)):
  File "/home/qinqian/anaconda3/envs/augm/lib/python3.8/site-packages/packaging/version.py", line 52, in parse
    return Version(version)
  File "/home/qinqian/anaconda3/envs/augm/lib/python3.8/site-packages/packaging/version.py", line 198, in __init__
    raise InvalidVersion(f"Invalid version: '{version}'")
packaging.version.InvalidVersion: Invalid version: '0.10.1,<0.11'

It seems that it is the transformer version problem, but I don't know how to fix it. Any reply will be appreciated. Thank you very much.

Best,
hangzhiyiwei

**Comment by JacobYuan7** [comment]
Well, can you tell me more about the environment that you use to run the code? The problem might be caused by the version of the transformer.

**Comment by hangzhiyiwei** [comment]
Thank you for your reply.  Here is the environment after running command "pip list"

Package                  Version
------------------------ ----------
asttokens                2.2.1
backcall                 0.2.0
certifi                  2023.5.7
charset-normalizer       3.1.0
click                    8.1.3
cloudpickle              2.2.1
cmake                    3.26.4
coloredlogs              15.0.1
contourpy                1.1.0
cycler                   0.11.0
Cython                   0.29.35
decorator                5.1.1
executing                1.2.0
filelock                 3.12.2
flatbuffers              23.5.26
fonttools                4.40.0
fsspec                   2023.6.0
h5py                     3.8.0
huggingface-hub          0.15.1
humanfriendly            10.0
idna                     3.4
importlib-resources      5.12.0
ipython                  8.12.2
jedi                     0.18.2
Jinja2                   3.1.2
joblib                   1.2.0
kiwisolver               1.4.4
lit                      16.0.6
MarkupSafe               2.1.3
matplotlib               3.7.1
matplotlib-inline        0.1.6
mpmath                   1.3.0
networkx                 3.1
numpy                    1.24.4
nvidia-cublas-cu11       11.10.3.66
nvidia-cuda-cupti-cu11   11.7.101
nvidia-cuda-nvrtc-cu11   11.7.99
nvidia-cuda-runtime-cu11 11.7.99
nvidia-cudnn-cu11        8.5.0.96
nvidia-cufft-cu11        10.9.0.58
nvidia-curand-cu11       10.2.10.91
nvidia-cusolver-cu11     11.4.0.1
nvidia-cusparse-cu11     11.7.4.91
nvidia-nccl-cu11         2.14.3
nvidia-nvtx-cu11         11.7.91
onnx                     1.14.0
onnxruntime              1.15.1
opencv-python            4.7.0.72
packaging                23.1
panopticapi              0.1
parso                    0.8.3
pexpect                  4.8.0
pickleshare              0.7.5
Pillow                   9.5.0
pip                      23.1.2
prettytable              3.8.0
prompt-toolkit           3.0.38
protobuf                 4.23.3
ptyprocess               0.7.0
pure-eval                0.2.2
pycocotools              2.0.2
Pygments                 2.15.1
pyparsing                3.1.0
python-dateutil          2.8.2
PyYAML                   6.0
regex                    2023.6.3
requests                 2.31.0
sacremoses               0.0.53
safetensors              0.3.1
scipy                    1.10.1
setuptools               67.8.0
six                      1.16.0
stack-data               0.6.2
submitit                 1.3.0
sympy                    1.12
timm                     0.9.2
tokenizers               0.10.3
torch                    2.0.1
torchvision              0.15.2
tqdm                     4.65.0
traitlets                5.9.0
transformers             4.5.1
triton                   2.0.0
typing_extensions        4.5.0
urllib3                  2.0.3
wcwidth                  0.2.6
wheel                    0.38.4
xmltodict                0.13.0
yacs                     0.1.8
zipp                     3.15.0

**Comment by JacobYuan7** [comment]
@hangzhiyiwei This is the environment that works.
**Hardware:** A100 (80GB)
**Software:** python 3.6
Package                       Version
----------------------------- --------
addict                        2.4.0
blis                          0.7.9
catalogue                     2.0.8
certifi                       2023.5.7
charset-normalizer            2.0.12
click                         8.0.4
clip                          1.0
cloudpickle                   2.2.1
confection                    0.0.4
contextvars                   2.4
cycler                        0.11.0
cymem                         2.0.7
Cython                        0.29.35
dataclasses                   0.8
en-core-web-sm                3.5.0
filelock                      3.4.1
ftfy                          6.0.3
huggingface-hub               0.4.0
idna                          3.4
immutables                    0.19
importlib-metadata            4.8.3
importlib-resources           5.4.0
Jinja2                        3.0.3
joblib                        1.1.1
kiwisolver                    1.3.1
langcodes                     3.3.0
MarkupSafe                    2.0.1
matplotlib                    3.3.4
mkl-fft                       1.3.0
mkl-random                    1.1.1
mkl-service                   2.3.0
mmcv                          1.7.1
MultiScaleDeformableAttention 1.0
murmurhash                    1.0.9
numpy                         1.19.5
olefile                       0.46
opencv-python                 4.7.0.72
packaging                     21.3
pathy                         0.10.1
Pillow                        8.3.1
pip                           21.2.2
preshed                       3.0.8
pycocotools                   2.0.2
pydantic                      1.9.2
pyparsing                     3.0.9
python-dateutil               2.8.2
PyYAML                        6.0
regex                         2023.5.5
requests                      2.27.1
sacremoses                    0.0.53
SceneGraphParser              0.1.0
scipy                         1.5.4
setuptools                    58.0.4
six                           1.16.0
smart-open                    6.3.0
spacy                         3.5.3
spacy-legacy                  3.0.12
spacy-loggers                 1.0.4
srsly                         2.4.6
submitit                      1.3.0
tabulate                      0.8.10
thinc                         8.1.10
timm                          0.6.12
tokenizers                    0.10.3
torch                         1.10.0
torchvision                   0.11.1
tqdm                          4.64.1
transformers                  4.5.1
typer                         0.7.0
typing_extensions             4.1.1
urllib3                       1.26.16
wasabi                        1.1.1
wcwidth                       0.2.6
wheel                         0.37.1
yapf                          0.32.0
zipp                          3.6.0

Well, I think the simplest way to make it run might be creating a new conda environment. This is the script that works for me on A100. You might need to check whether pytorch, torchvision and cudatoolkit can work on your computer.

```
conda create -n rlip python=3.6 -y
conda init
conda activate rlip
conda install pytorch==1.10.0 torchvision==0.11.1 cudatoolkit=11.3 -c pytorch -y
pip install opencv-python
pip install scipy
```

**Comment by hangzhiyiwei** [comment]
Thank you for your reply. I found that it is the problem of the "packaging". It should be the 21.3 version.
'''
pip uninstall packaging
pip install packaging=21.3
'''
